atc/metric/metrics.go

```diff
@@ -111,24 +111,51 @@ func (event SchedulingJobDuration) Emit(logger lager.Logger) {
 }
 
 type WorkerContainers struct {
-	WorkerName string
-	Platform   string
-	Containers int
+	WorkerName       string
+	Platform         string
+	ActiveContainers int
+	BuildContainers  int
+	CheckContainers  int
 }
 
 func (event WorkerContainers) Emit(logger lager.Logger) {
 	emit(
 		logger.Session("worker-containers"),
 		Event{
 			Name:  "worker containers",
-			Value: event.Containers,
+			Value: event.ActiveContainers,
 			State: EventStateOK,
 			Attributes: map[string]string{
 				"worker":   event.WorkerName,
 				"platform": event.Platform,
 			},
 		},
 	)
+
+	emit(
+		logger.Session("build-containers-count"),
+		Event{
+			Name:  "build containers",
+			Value: event.BuildContainers,
+			State: EventStateOK,
+			Attributes: map[string]string{
+				"worker": event.WorkerName,
+			},
+		},
+	)
+
+	emit(
```

Comment:
Hey,

Given that for `prometheus` we still need to specially treat the metrics in the emitter, it'd be nice to have a case for check containers https://github.com/concourse/concourse/blob/2c374aaaf6ebc619df4439ca6d21c68a59c77fde/atc/metric/emitter/prometheus.go#L333 (or did I miss something?)

Thx!